pkg/kafka/subscriber.go

```diff
@@ -578,6 +581,9 @@ ResendLoop:
 		case <-ctx.Done():
 			h.logger.Trace("Closing, ctx cancelled before sent to consumer", receivedMsgLogFields)
 			return nil
+		case <-sessionContext.Done():
```

Comment:
won't this fail since `sess` is nil?

Reply:
Session can't be nil. Saram won't start consuming in such case. By the way sarama does the same check before consuming starts, but if rebalancing started after, during processing messages, it hangs on trying commit offset with staled session.

Reply:
What about this then? https://github.com/ThreeDotsLabs/watermill-kafka/blob/master/pkg/kafka/subscriber.go#L452

Reply:
Thanks. I've overlooked it. Fixed this.